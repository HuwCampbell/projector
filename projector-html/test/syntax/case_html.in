{
  case foo of
    bar ->
      <blink>party</blink>
    baz ->
      <marquee>cube</marquee>
      <i>abcdefg</i>
    Bap ->
      <b>hell</b>
      <i>foo</i>
}