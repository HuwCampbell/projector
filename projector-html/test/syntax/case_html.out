Template
  ()
  Nothing
  (TENode
     ()
     (THtml
        ()
        [ TExprNode
            ()
            (TECase
               ()
               (TEVar () TId { unTId = "foo" })
               (TAlt
                  ()
                  (TPVar () TId { unTId = "bar" })
                  (TENode
                     ()
                     (THtml
                        ()
                        [ TElement
                            ()
                            (TTag () "blink")
                            []
                            (THtml () [ TPlain () TPlainText { unTPlainText = "party" } ])
                        , TWhiteSpace ()
                        ])) :|
                  [ TAlt
                      ()
                      (TPVar () TId { unTId = "baz" })
                      (TENode
                         ()
                         (THtml
                            ()
                            [ TWhiteSpace ()
                            , TElement
                                ()
                                (TTag () "marquee")
                                []
                                (THtml () [ TPlain () TPlainText { unTPlainText = "cube" } ])
                            , TWhiteSpace ()
                            , TElement
                                ()
                                (TTag () "i")
                                []
                                (THtml () [ TPlain () TPlainText { unTPlainText = "abcdefg" } ])
                            , TWhiteSpace ()
                            ]))
                  , TAlt
                      ()
                      (TPCon () TConstructor { unTConstructor = "Bap" } [])
                      (TENode
                         ()
                         (THtml
                            ()
                            [ TWhiteSpace ()
                            , TElement
                                ()
                                (TTag () "b")
                                []
                                (THtml () [ TPlain () TPlainText { unTPlainText = "hell" } ])
                            , TWhiteSpace ()
                            , TElement
                                ()
                                (TTag () "i")
                                []
                                (THtml () [ TPlain () TPlainText { unTPlainText = "foo" } ])
                            , TWhiteSpace ()
                            ]))
                  ]))
        ]))